client/components/main/Desktop/TypeSelector/index.js

```diff
@@ -27,26 +28,44 @@ const useHeaderStyles = makeStyles(theme => ({
   }
 }))
 
+
+
+
 const RequestTypeSelector = ({
   requestTypes,
   updateTypesFilter,
   selectedTypes,
 }) => {
-  const [leftCol, setLeftCol] = useState();
-  const [rightCol, setRightCol] = useState();
+  // const [leftCol, setLeftCol] = useState();
+  // const [rightCol, setRightCol] = useState();
+  const [items, setCols] = useState();
   const classes = useStyles();
   const headerClass = useHeaderStyles();
+  const [isToggled, toggle] = useToggle(false);
 
   useEffect(() => {
     if (requestTypes) {
-      const mid = Math.ceil(requestTypes.length / 2);
-      const left = requestTypes.slice(0, mid);
-      const right = requestTypes.slice(-mid);
-      setLeftCol(left);
-      setRightCol(right);
+      // const mid = Math.ceil(requestTypes.length / 2);
+      // const left = requestTypes.slice(0, mid);
+      // const right = requestTypes.slice(-mid);
+      // setLeftCol(left);
+      // setRightCol(right);
+
+      const items = requestTypes;
+      setCols(items);
     }
   }, [requestTypes]);
 
+  const checkAll = (items) => {
+     toggle(!isToggled);
+     if(!isToggled){
+      items.map(type=>selectedTypes[type.typeId] = true);
+     }else{
+      items.map(type=>selectedTypes[type.typeId] = false);
```

Comment:
Since `selectedTypes` is a prop we're mapping from the redux store, we should not directly mutate `selectedTypes`. Instead, you'd want to dispatch an action via a function in `mapDispatchToProps`. Let me know if you've got any questions!